kernel/meson.build

```diff
@@ -1175,6 +1174,20 @@ foreach _kop : base_kops
           endforeach
         endif
 
+        # ?{hemm,symm}3m_* Rules
+        if '3m_' in ext
+          foreach sw_key, sw_val : m3_startswith
+            if ext.startswith(sw_key)
+              _ext_cargs += sw_val
+            endif
+          endforeach
+          foreach ew_key, ew_val : m3_endswith
+            if ext.endswith(ew_key)
+              _ext_cargs += ew_val
+            endif
+          endforeach
+        endif
```

Comment:
@mattip this is how the defines are generalized in `meson` (no functions, so `foreach` only).